flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/StreamTaskTest.java

```diff
@@ -410,6 +410,42 @@ public CloseException() {
         }
     }
 
+    @Test
+    public void testAsyncExceptionCanBeHandledDuringRestoring() throws Exception {
+        MockEnvironment mockEnvironment = MockEnvironment.builder().build();
+        RuntimeException expectedException = new RuntimeException("RUNTIME EXCEPTION");
+
+        mockEnvironment.setExpectedExternalFailureCause(AsynchronousException.class);
+        final String expectedErrorMessage = "EXPECTED_ERROR MESSAGE";
+
+        StreamTaskITCase.NoOpStreamTask noOpStreamTask =
```

Comment:
This is indeed not no-op, maybe should call it `initThrowExceptionTask` or other more appropriate name

Reply:
`initThrowExceptionTask` is appropriate. a good proposal.👍